tests/test_transmissionCurve.py

```diff
@@ -0,0 +1,554 @@
+            useOutArg(tc.sampleAt, wavelengths.size, self.point, tc.getNaturalSampling()),
+            rtol=1E-11, atol=1E-11
+        )
+        self.checkPersistence(tc)
+
+    def testConstantEvenSpacing(self):
+        # Don't want to accidentally test floating-point ops with numbers
+        # that can be exactly represented, so we add random values
+        minWavelength = 6000 + self.random.randn()
+        maxWavelength = 6200 + self.random.randn()
+        nan = float("nan")
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, 0.0, 0.0)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, 0.0, 1.0)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, 0.0, nan)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, 1.0, 0.0)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, 1.0, 1.0)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, 1.0, nan)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, nan, 0.0)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, nan, 1.0)
+        self.checkConstantEvenSpacing(minWavelength, maxWavelength, nan, nan)
+
+    def checkConstantUnevenSpacing(self, minWavelength, maxWavelength, throughputAtMin, throughputAtMax):
+        """Test that we can construct and use a spatially-constant
+        TransmissionCurve initialized with an unevenly-spaced wavelength
+        array.
+        """
+        wavelengths = minWavelength + (maxWavelength - minWavelength)*self.random.rand(100)
+        wavelengths.sort()
+        curve = makeTestCurve(self.random, minWavelength, maxWavelength, throughputAtMin, throughputAtMax)
+        tc = self.makeAndCheckConstant(curve, wavelengths, throughputAtMin, throughputAtMax)
+        self.assertLess(tc.getNaturalSampling().getSpacing(), min(wavelengths[1:] - wavelengths[:-1]))
+        self.checkPersistence(tc)
+
+    def testConstantUnevenSpacing(self):
+        # Don't want to accidentally test floating-point ops with numbers
+        # that can be exactly represented, so we add random values
+        minWavelength = 7200 + self.random.randn()
+        maxWavelength = 7400 + self.random.randn()
+        nan = float("nan")
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, 0.0, 0.0)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, 0.0, 1.0)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, 0.0, nan)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, 1.0, 0.0)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, 1.0, 1.0)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, 1.0, nan)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, nan, 0.0)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, nan, 1.0)
+        self.checkConstantUnevenSpacing(minWavelength, maxWavelength, nan, nan)
```

Comment:
I'm not sure I understand your suggestion; I don't want to use any kind of iteration because that hides what case failed in any failure traceback, but it's quite possible there's a `unittest` feature I'm not aware of to deal with this.

Reply:
(Just found a later comment with what I think is a link to an example that might apply to this too; will look into it).

Reply:
For the record: unittest asserts accept a "msg" argument that can be used to indicate which inputs failed in a loop.  If you sent self.longmessage = True then you also see the usual message  explaining what was wrong (strongly recommended).

unittest in Python 3.4 has a new context manager subTest that makes all this much simpler, but we can't use it quite yet.